### PR TITLE
separate tracker and pool concerns

### PR DESCRIPTION
**Pull request by emergent-reasons**
I am assuming this will be rejected as unnecessary or too late. That's fine. I just wanted to see what it looks like with the tracker and pool concerns separated and I thought you might like to see it too.

Note: this is based on the branch from #51 so it is a little hard to see the specific changes unless #51 gets merged.

 It is mostly shuffling things around. There are a few changes:

- overall I think the code is easier to read this way
- frozen pool does not need to be tracked separately so I just checked for the snapshot
- It removed one or two re-entrant lock issues so the method signatures could be simplified.
- Pool has its own mutex but it should be redundant with the tracker since I did not attempt to change any tracker logic.

### CODE REVIEW

**Comment by zquestz** [comment]
This actually looks great. I think this cleanup makes a lot of sense. The only part I am not crazy about is the naming of `IsFrozen` perhaps it should just be `IsReady` or `IsFull`?

File: server/pool.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	defer pool.mutex.Unlock()
+
+	delete(pool.players, player.number)
+}
+
+// PlayerFromSnapshot returns the user for the key or nil if they
+// are not in the snapshot.
+func (pool *Pool) PlayerFromSnapshot(verificationKey string) *playerData {
+	pool.mutex.RLock()
+	pool.mutex.RUnlock()
```
Comment (zquestz) [inline server/pool.go]: Missing defer.

Comment (emergent-reasons) [reply server/pool.go]: Thanks! b9bf12c

**Comment by emergent-reasons** [comment]
I merged from `unique-verification-keys` since that is more important and should land first.

**Comment by zquestz** [comment]
I believe this can be closed since the other PR landed.

**Comment by emergent-reasons** [comment]
Yes. Thanks a lot for reviewing everything.

**Pull request closed without merging**